src/core/io/NclReader.cpp

```diff
@@ -1919,9 +1919,6 @@ Tree* NclReader::translateNclSimpleTreeToBranchLengthTree(NxsSimpleTree& nTree,
         tau->getNode(nodes[i]->getIndex()).setBranchLength( brlens[i] );
     }
     
-    bool fossils_only = true;
```

Comment:
why is this removed? I would rather make all internal nodes bifurcating and work with 0-length branches. Our code, such as PhyloCTMC, always assumes bifurcating nodes.

Reply:
Not modifying trees while we are reading them was one of the main goals of this PR.

Modifying the trees while we are reading them has caused several bugs so far.  First, if reindex == false, then either
* the tree contains no "knuckles" and we didn't do anything.
* the tree contains "knuckles" and we crashed because they don't have a node index.

So, I changed reindex to be always `true`, and then this changed the `index` attribute in some cases, which caused problems with ancestral state reconstruction, and with loading trees from a checking.

Overall, I think we should not be modifying trees while reading them.

Its fine for phyloctmc to throw an exception if it sees a tree that it doesn't like.

Reply:
OK, still seems weird to me that we may be inconsistent internally in sometimes allowing for degree-1 nodes and sometimes allowing 0-length branches. This really should be made consistent; for writing, reading and internal use.